### PR TITLE
Removed HOCs and replaced completely with hooks

### PR DESCRIPTION
**Pull request by justin0022**
* The higher order components `withResponsiveness` and `createChartComponent` were replaced with hooks `useResponsiveness` and `useCreateChart`
* `el` was renamed to `domElement` to improve readability.

### CODE REVIEW

**Comment by pushyamig** [comment]
Here are my views on this PR, Some of the code that is refactored here are the JS file that we don't use it in the project like `LineChart.js` `HorizontalBarChart.js`, `ScatterPlot.js` etc. Since we are not using it we need to remove from the project. Just as we removed `withResponsiveness.js` and `createChartComponenet.js`. It creates confusion for somebody new coming into the project these files there if not using. If the worry the work being lost then you can store in a local forked branch and pull it up when needed.  If the ideas about the code maintenance are different among others we can have a discussion on this in our next meeting

**Comment by justin0022** [comment]
I'm happy to remove unused D3 components. I have them stored in a separate repo anyway since I use them for other projects.

File: assets/src/components/BarChart.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-import withResponsiveness from './withResponsiveness'
-import createChartComponent from './createChartComponent'
-import compose from '../util/compose'
-
-const BarChart = compose(
-  withResponsiveness,
-  createChartComponent
-)(createBarChart)
+import useResponsiveness from '../hooks/useResponsiveness'
+import useCreateChart from '../hooks/useCreateChart'
```
Comment (pushyamig) [inline assets/src/components/BarChart.js]: This can be deleted as well as we are not using it either for Assignment view or files view

File: assets/src/components/d3/createLineChart.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ function createLineChart ({ data, width, height, el }) {
     .attr('transform', `translate(${margin.left},0)`)
     .call(d3.axisLeft(y))
 
-  const svg = createSVG({ el, width, height })
+  const svg = createSVG({ domElement, width, height })
```
Comment (pushyamig) [inline assets/src/components/d3/createLineChart.js]: This can be deleted as well

File: assets/src/components/BarChart.js (hunk 1 above)
Comment (justin0022) [reply assets/src/components/BarChart.js]: Okay I've removed this one as well

File: assets/src/components/d3/createLineChart.js (hunk 2 above)
Comment (justin0022) [reply assets/src/components/d3/createLineChart.js]: This one has been removed

File: assets/src/components/BarChart.js (hunk 1 above)
Comment (justin0022) [reply assets/src/components/BarChart.js]: I've deleted these now

File: assets/src/components/d3/createLineChart.js (hunk 2 above)
Comment (justin0022) [reply assets/src/components/d3/createLineChart.js]: I've deleted these now

File: assets/src/containers/App.js [hunk 3]
```diff
@@ -18,14 +22,24 @@ function App () {
...
+          sideDrawerState={sideDrawerState}
+          courseId={courseId}
+          courseInfo={loaded ? courseInfo : null} />
+        <Route path='/:courseId/' exact
+          render={props => <IndexPage {...props} courseInfo={loaded ? courseInfo : null} courseId={courseId} />} />
+        <Route path='/:courseId/grades' render={props => <GradeDistribution {...props} courseId={courseId} />} />
+        <Route path='/:courseId/assignment' render={props => <AssignmentPlanning {...props} courseId={courseId} />} />
+        <Route path='/:courseId/files' render={props => <FilesAccessed {...props} courseId={courseId} />} />
+      </>
     </Router>
```
Comment (pushyamig) [inline assets/src/containers/App.js]: How is this commit relevant here? this PR is addressing removing HOC with hooks whereas this looks like calling hooks for getting courseInfo and some other changes based on that call.

isn't this supposed to be [PR#481](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/481)

Comment (justin0022) [reply assets/src/containers/App.js]: Sorry, I think this was a mistake, I'll revert the `App.js` back to how it was.
